### PR TITLE
Add detail on state mismatch

### PR DESCRIPTION
**Pull request by achamayou**
Doing a pass on tickets, #204 seemed overdue and I expect will be useful shortly.

### CODE REVIEW

File: src/node/nodestate.h [hunk 1]
```diff
@@ -61,8 +102,11 @@ namespace ccf
     StateMachine(T s) : s(s) {}
     void expect(T s) const
     {
-      if (s != this->s.load())
-        throw std::logic_error("Unexpected state");
+      auto state = this->s.load();
+      if (s != state)
+      {
+        throw std::logic_error(fmt::format("Unexpected state: {}", state));
```
Comment (eddyashton) [inline src/node/nodestate.h]: Message should include the _expected_ state as well as the current state.

Comment (achamayou) [reply src/node/nodestate.h]: That's much easier to infer from the trace, but sure, why not!

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**